src/bun.js/webcore/blob.zig

```diff
@@ -109,6 +109,10 @@ pub const Blob = struct {
     globalThis: *JSGlobalObject = undefined,
 
     last_modified: f64 = 0.0,
+    /// Blob name will lazy initialize when getName is called, but
+    /// we must be able to set the name, and we need to keep the value alive
+    /// https://github.com/oven-sh/bun/issues/10178
+    name: JSC.Strong = .{},
```

Comment:
If we ensure the name is only allowed to be a string, can't we get away with a `bun.String` here instead? Otherwise, couldn't we use a `mutable WriteBarrier` (nameGetCachedValue)?